packages/front-end/components/Experiment/StartExperimentBanner.tsx

```diff
@@ -218,9 +288,58 @@ export function StartExperimentBanner({
     onStart && onStart();
   }
 
-  const allPassed = !checklist.some((c) => c.status === "error");
+  const isTaskCompleted = (currentTask: string) => {
+    const index = manualChecklistStatus.findIndex(
+      (task) => task.key === currentTask
+    );
+
+    if (index === -1 || !manualChecklistStatus[index]) {
+      return false;
+    }
+
+    return manualChecklistStatus[index].status === "complete";
+  };
+
+  const allPassed =
+    !checklist.some((c) => c.status === "error") &&
+    manualChecklist.every((task) => isTaskCompleted(task.key));
+
+  async function updateTaskStatus(checked: boolean, item: ManualChecklist) {
+    const updatedManualChecklistStatus = Array.isArray(manualChecklistStatus)
+      ? [...manualChecklistStatus]
+      : [];
+
+    if (!updatedManualChecklistStatus.length) {
```

Comment:
Singling out this empty array case makes this code harder to read.  You can delete this and just keep the part inside `else` and it will work fine (`findIndex` will return -1 if the array is empty)